gpflow/base.py

```diff
@@ -320,13 +344,15 @@ def _cast_to_dtype(value: VariableData, dtype: Optional[DType] = None) -> tf.Ten
         return tf.convert_to_tensor(value, dtype=dtype)
 
 
-def _to_constrained(value: VariableData, transform: Transform) -> tf.Tensor:
+def _to_constrained(value: tf.Tensor, transform: Optional[Transform]) -> tf.Tensor:
     if transform is not None:
         return transform.forward(value)
     return value
 
 
-def _to_unconstrained(value: VariableData, transform: Transform) -> tf.Tensor:
+def _to_unconstrained(
+        value: Union[tf.Tensor, tf.Variable], transform: Optional[Transform]
```

Comment:
can take any TensorLike

Reply:
yes, this one can take a `TensorLike`. odd - i thought it wouldn't work in `Parameter`, but hey. I've also corrected `validate_unconstrained_value` to return just `tf.Tensor`